### ISSUE TITLE
clang-tidy misreports indentation with tabs and spaces

### ISSUE THREAD

**Issue by orangebowlerhat** [clangd/clangd#1987]
When a piece of code contains and `if-else` statement, clang-tidy reports that the `if` and `else` are not at the same indentation and that's a great idea. However, clang-tidy sometimes reports the non-matching indent on `if-else` where the indentation is correct.

I'm using tabs for indents, 4 characters wide. The `if` and `else` have the same number of tabs. However the `else` had a stray space after one of the tabs. This does not change the indentation at all, but clang-tidy reports it as a mismatch.

AFAIK there is no case in which the space could cause the indentation not to match. Tabs have to be at least one character wide.

clangd version 16.0.6
Features: linux
Platform: x86_64-pc-linux-gnu

I'm using VSCode and I understand clang-tidy is run by its cpp extension. Although the clangd extension is installed and enabled also. 

Manjaro Linux, XFCE

**Comment by HighCommander4** [comment]
Can you provide a code example where you get the unexpected diagnostic?

**Comment by orangebowlerhat** [comment]
Absolutely, I would have done already but I wasn't sure how the indentation would work on Github.
```
		if ((knot->state & kntTYPE) != kntLINE) {
			gxMark (x + knot->length / 3.f, knot->out.z, 0.f, gxMARK_GEM, 0xff00ffff);
			x += knot->length;
			gxMark (x - knot->length / 3.f, knot->next->in.z, 0.f, gxMARK_GEM, 0xff00ffff);
		}
	 	else x += knot->length;
```
Note that, on the `else` line, there is a space before the second tab character. This reports -

`Different indentation for 'if' and corresponding 'else'clang-tidy[readability-misleading-indentation](https://clang.llvm.org/extra/clang-tidy/checks/readability/misleading-indentation.html)
`
I notice now that its the VSCode cpp plugin formatting option, but clearly the message originates in clang-tidy. While the indentation characters aren't exactly the same, the result isn't misleading.

**Comment by HighCommander4** [comment]
Thanks. I can reproduce this on a minimal example with `clang-tidy` on the command line:

```c++
int foo(bool b) {
	if (b)
		return 42;
 	else
		return 0;
}
```

```console
$ clang-tidy --checks=readability-misleading-indentation test.cpp
1 warning generated.
test.cpp:4:3: warning: different indentation for 'if' and corresponding 'else' [readability-misleading-indentation]
    4 |         else
      |         ^
```

This tells us that this is a behaviour of the clang-tidy check itself. Clangd just passes through the diagnostic to the user.

If you believe the behaviour is incorrect, please file an upstream clang-tidy issue at https://github.com/llvm/llvm-project/issues/. Thanks!

**Comment by HighCommander4** [comment]
Closing as duplicate of the upstream issue.
